core/controllers/editor.py

```diff
@@ -122,11 +123,8 @@ def get(self, exploration_id):
         exploration = exp_services.get_exploration_by_id(
             exploration_id, strict=False)
 
-        can_edit = (
-            bool(self.user_id) and
-            self.username not in config_domain.BANNED_USERNAMES.value and
-            rights_manager.Actor(self.user_id).can_edit(
-                feconf.ACTIVITY_TYPE_EXPLORATION, exploration_id))
+        exploration_rights = rights_manager.get_exploration_rights(
```

Comment:
Again see if you can find a way to batch this with retrieving the exploration. Though if you like you can defer this (batching calls) to a separate PR, as it will probably require some refactoring.